python_modules/dagster/dagster/_config/field_utils.py

```diff
@@ -474,7 +474,11 @@ def config_dictionary_from_values(
     return check.is_dict(_config_value_to_dict_representation(None, values))
 
 
-class IntEnvVar(int):
+class DagsterEnvVar:
```

Comment:
Minor nit: I don't think this should be a superclass but instead 

DagsterEnvVar = Union[IntEnvVar, EnvVar]